controllers/kustomization_gc.go

```diff
@@ -120,8 +122,13 @@ func (kgc *KustomizeGarbageCollector) Prune(timeout time.Duration, name string,
 	return changeSet, true
 }
 
-func (kgc *KustomizeGarbageCollector) matchingLabels(name, namespace, checksum string) client.MatchingLabels {
-	return gcLabels(name, namespace, checksum)
+func (kgc *KustomizeGarbageCollector) isStale(obj unstructured.Unstructured) bool {
+	itemChecksum := obj.GetLabels()[fmt.Sprintf("%s/checksum", kustomizev1.GroupVersion.Group)]
```

Comment:
I would move the `fmt.Sprintf` to a dedicated function as it is also used in the `gcLabels` function, or move the various label names to constants, and create something like:

```go
const gcNameLabel = "name"

func gcLabelName(name string) string {
	return fmt.Sprintf("%s/%s", kustomizev1.GroupVersion.Group, name)
}
```